### ISSUE TITLE
Rule Change: (object-shorthand) ignore confusing globals

### ISSUE THREAD

**Issue by maxpatiiuk** [eslint/eslint#17894]
### What rule do you want to change?

object-shorthand

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
const a = { undefined: undefined };
const b = { Infinity: Infinity };
const c = { NaN: NaN };
```

### What does the rule currently do for this code?

```js
const a = { undefined };
const b = { Infinity };
const c = { NaN };
```

### What will the rule do after it's changed?

```js
const a = { undefined: undefined };
const b = { Infinity: Infinity };
const c = { NaN: NaN };
```

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

Using object-shorthand syntax when it comes to global variables can result in confusing code/confusing behavior (i.e I don't have an "Infinity" variable defined in my file so why did object shorthand get used?)

it's not obvious that `const a = { undefined }` is equivalent to `const a = { undefined: undefined }`
This is especially confusing for less common globals

I propose:
- Either add an option to exclude built-in globals from object-shorthand rule
- Or, exclude built-in globals from object-shorthand by default
- Or add an option for manually providing exclude list (just like there is methodsIgnorePattern, add propsIgnorePattern)

**Comment by mdjermanovic** [comment]
Hi @maxpatiiuk, thanks for the issue!

Per our rules policies, [stylistic rules are frozen](https://github.com/eslint/eslint#stylistic-rule-updates) so we are not adding new options to the `object-shorthand` rule. You can always copy the existing rule into a custom rule and make it behave the way you want.
